src/clarity.rs

```diff
@@ -366,45 +411,53 @@ pub fn invoke_command(invoked_by: &str, args: &[String]) {
-                eprintln!("Usage: {} {} [contract-identifier] (program.clar) [vm-state.db]", invoked_by, args[0]);
-                panic_test!();
-            }
-
+            let evalInput = eval(invoked_by, args);
             let vm_filename = 
                 if args.len() == 3 {
                     &args[2]
                 } else {
                     &args[3]
                 };
-
-
-
-            let content: String = {
+            let marf_kv = friendly_expect(MarfedKV::open(vm_filename, None), "Failed to open VM database.");
+            let result = in_block(vm_filename, marf_kv, |mut marf| {
+                let result = {
+                    let db = marf.as_clarity_db(&NULL_HEADER_DB);
+                    let mut vm_env = OwnedEnvironment::new_cost_limited(db, LimitedCostTracker::new_max_limit());
+                    vm_env.get_exec_environment(None)
+                        .eval_read_only(&evalInput.contract_identifier, &evalInput.content)
+                };
+                (marf, result)
+            });
+        
+            match result {
+                Ok(x) => {
+                    println!("Program executed successfully! Output: \n{}", x);
+                },
+                Err(error) => { 
+                    eprintln!("Program execution error: \n{}", error);
+                    panic_test!();
+                }
+            }
+        },
+        "eval_at_chaintip" => {
+            let evalInput = eval(invoked_by, args);
+            let vm_filename = 
                 if args.len() == 3 {
-                    let mut buffer = String::new();
-                    friendly_expect(io::stdin().read_to_string(&mut buffer),
-                                    "Error reading from stdin.");
-                    buffer
+                    &args[2]
                 } else {
-                    friendly_expect(fs::read_to_string(&args[2]),
-                                    &format!("Error reading file: {}", args[2]))
-                }
-            };
-
-            let contract_identifier = friendly_expect(QualifiedContractIdentifier::parse(&args[1]), "Failed to parse contract identifier.");
-
+                    &args[3]
+                };
             let marf_kv = friendly_expect(MarfedKV::open(vm_filename, None), "Failed to open VM database.");
-            let result = in_block(vm_filename, marf_kv, |mut marf| {
+            let result = at_chaintip(vm_filename, marf_kv, |mut marf| {
                 let result = {
                     let db = marf.as_clarity_db(&NULL_HEADER_DB);
                     let mut vm_env = OwnedEnvironment::new_cost_limited(db, LimitedCostTracker::new_max_limit());
                     vm_env.get_exec_environment(None)
-                        .eval_read_only(&contract_identifier, &content)
+                        .eval_read_only(&evalInput.contract_identifier, &evalInput.content)
                 };
                 (marf, result)
             });
-
+        
```

Comment:
I don't see any new unit tests for `eval_at_chaintip`.  Can you add some?  See the `test_samples()` unit test battery below this.

Reply:
Added some tests - I made a new file for the program instead of using stdin.

Reply:
Thanks!